Crashing when downloading because of SAF

<!--
Oh no, a bug! It happens. Thanks for reporting an issue with NewPipe. To make it easier for us to help you please enter detailed information in the template we have provided below. If a section isn't relevant, just delete it, though it would be helpful to still provide as much detail as possible.
-->

<!-- IF YOU DON'T FILL IN THE TEMPLATE PROPERLY, YOUR ISSUE IS LIABLE TO BE CLOSED. If you feel tired/lazy right now, open your issue some other time. We'll wait. -->

<!-- The comments between these brackets won't show up in the submitted issue (as you can see in the Preview). -->

### Checklist
<!-- This checklist is COMPULSORY. The first box has been checked for you to show you how it is done. -->

- [x] I am using the latest version - x.xx.x <!-- Check https://github.com/TeamNewPipe/NewPipe/releases -->
- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. <!-- Seriously, check. O_O -->
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one bug. I will open one issue for every bug report I want to file.

### Steps to reproduce the bug
<!--
1. Go to 'any video'
2. Press on 'download'

-->
download any video or go to see download folder in the app, app restarts a couple of time and crashes

happening only if I enable SAF, its default in 0.21.6 also happens in 0.21.5 if I enabled it

but in 0.21.5, I could atleast see my downloads folder, in 0.21.6 it crashes in both cases, seeing the downloads folder or downloading a video/audio

<!-- If you can't cause the bug to show up again reliably (and hence don't have a proper set of steps to give us), please still try to give as many details as possible on how you think you encountered the bug. -->



### Actual behavior
<!-- Tell us what happens with the steps given above. -->



### Expected behavior
<!-- Tell us what you expect to happen. -->



### Screenshots/Screen recordings
<!-- If applicable, add screenshots or a screen recording to help explain your problem. GitHub supports uploading them directly in the issue text box. If your file is too big for Github to accept, feel free to paste a link from an image/video hoster here instead. -->

<!-- DON'T POST SCREENSHOTS OF THE ERROR PAGE. Use the buttons given on the error page to paste the error as text in the Logs section below. -->


https://user-images.githubusercontent.com/84063610/126030104-584b3d52-d456-4769-9c7d-dbb55cd1a863.mp4



### Logs
<!-- If your bug includes a crash (where you're shown the Error Report page with a bunch of info), tap on "Copy formatted report" at the bottom and paste it here: -->

<!-- That's right, here! -->

`## Exception
* __User Action:__ ui error
* __Request:__ ACRA report
* __Content Country:__ US
* __Content Language:__ en-US
* __App Language:__ en_US
* __Service:__ none
* __Version:__ 0.21.6
* __OS:__ Linux Android 10 - 29
<details><summary><b>Crash log </b></summary><p>

```
java.lang.RuntimeException: Unable to create service us.shandian.giga.service.DownloadManagerService: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String org.schabi.newpipe.streams.io.StoredFileHelper.toString()' on a null object reference
	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4412)
	at android.app.ActivityThread.access$1800(ActivityThread.java:274)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2118)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loop(Looper.java:237)
	at android.app.ActivityThread.main(ActivityThread.java:8167)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String org.schabi.newpipe.streams.io.StoredFileHelper.toString()' on a null object reference
	at us.shandian.giga.service.DownloadManager.loadPendingMissions(DownloadManager.java:165)
	at us.shandian.giga.service.DownloadManager.<init>(DownloadManager.java:77)
	at us.shandian.giga.service.DownloadManagerService.onCreate(DownloadManagerService.java:142)
	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4400)
	... 8 more

```
</details>
<hr>
`

<!-- Please fill this out when you do not provide a log generate by NewPipe -->

### Device info

 - Android version/Custom ROM version:
 - Device model:

Thanks! I tried to disable SAF, but I still get the same error:

## Exception
* __User Action:__ ui error
* __Request:__ ACRA report
* __Content Country:__ GB
* __Content Language:__ en
* __App Language:__ en_US
* __Service:__ none
* __Version:__ 0.21.6
* __OS:__ Linux Android 9 - 28
<details><summary><b>Crash log </b></summary><p>

```
java.lang.RuntimeException: Unable to create service us.shandian.giga.service.DownloadManagerService: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String org.schabi.newpipe.streams.io.StoredFileHelper.toString()' on a null object reference
	at android.app.ActivityThread.handleCreateService(ActivityThread.java:3579)
	at android.app.ActivityThread.access$1300(ActivityThread.java:197)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1669)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:193)
	at android.app.ActivityThread.main(ActivityThread.java:6671)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:495)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:860)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String org.schabi.newpipe.streams.io.StoredFileHelper.toString()' on a null object reference
	at us.shandian.giga.service.DownloadManager.loadPendingMissions(DownloadManager.java:165)
	at us.shandian.giga.service.DownloadManager.<init>(DownloadManager.java:77)
	at us.shandian.giga.service.DownloadManagerService.onCreate(DownloadManagerService.java:142)
	at android.app.ActivityThread.handleCreateService(ActivityThread.java:3567)
	... 8 more

```
</details>

Same issue for me. I'm on Android 11 so no option to disable.

`## Exception
* __User Action:__ ui error
* __Request:__ ACRA report
* __Content Country:__ GB
* __Content Language:__ en-GB
* __App Language:__ en
* __Service:__ none
* __Version:__ 0.21.6
* __OS:__ Linux Android 11 - 30
<details><summary><b>Crash log </b></summary><p>

```
java.lang.RuntimeException: Unable to create service us.shandian.giga.service.DownloadManagerService: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String org.schabi.newpipe.streams.io.StoredFileHelper.toString()' on a null object reference
	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4198)
	at android.app.ActivityThread.access$1500(ActivityThread.java:237)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1932)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:223)
	at android.app.ActivityThread.main(ActivityThread.java:7660)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String org.schabi.newpipe.streams.io.StoredFileHelper.toString()' on a null object reference
	at us.shandian.giga.service.DownloadManager.loadPendingMissions(DownloadManager.java:165)
	at us.shandian.giga.service.DownloadManager.<init>(DownloadManager.java:77)
	at us.shandian.giga.service.DownloadManagerService.onCreate(DownloadManagerService.java:142)
	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4186)
	... 8 more

```
</details>
<hr>
`

Issue remains on version 0.21.7

`## Exception
* __User Action:__ ui error
* __Request:__ ACRA report
* __Content Country:__ GB
* __Content Language:__ en-GB
* __App Language:__ en
* __Service:__ none
* __Version:__ 0.21.7
* __OS:__ Linux Android 11 - 30
<details><summary><b>Crash log </b></summary><p>

```
java.lang.RuntimeException: Unable to create service us.shandian.giga.service.DownloadManagerService: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String org.schabi.newpipe.streams.io.StoredFileHelper.toString()' on a null object reference
	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4198)
	at android.app.ActivityThread.access$1500(ActivityThread.java:237)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1932)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:223)
	at android.app.ActivityThread.main(ActivityThread.java:7660)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String org.schabi.newpipe.streams.io.StoredFileHelper.toString()' on a null object reference
	at us.shandian.giga.service.DownloadManager.loadPendingMissions(DownloadManager.java:165)
	at us.shandian.giga.service.DownloadManager.<init>(DownloadManager.java:77)
	at us.shandian.giga.service.DownloadManagerService.onCreate(DownloadManagerService.java:142)
	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4186)
	... 8 more

```
</details>
<hr>
`

--- a/app/src/main/java/us/shandian/giga/service/DownloadManager.java
+++ b/app/src/main/java/us/shandian/giga/service/DownloadManager.java
@@ -149,7 +149,7 @@ public class DownloadManager {
             if (sub.getName().equals(".tmp")) continue;
 
             DownloadMission mis = Utility.readFromFile(sub);
-            if (mis == null || mis.isFinished()) {
+            if (mis == null || mis.isFinished() || mis.hasInvalidStorage()) {
                 //noinspection ResultOfMethodCallIgnored
                 sub.delete();
                 continue;

